hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MethodProbe.java

```diff
@@ -93,61 +173,68 @@ static <S> MethodProbe createMethodProbe(Method method, Probe probe, SourceMetad
-
-        LongMethodProbe(Method method, Probe probe, int type, SourceMetadata sourceMetadata) {
+        LongMethodProbe(final Method method, final Probe probe, final ProbeType type, final SourceMetadata sourceMetadata) {
             super(method, probe, type, sourceMetadata);
         }
 
+        @SuppressWarnings("checkstyle:CyclomaticComplexity")
         @Override
-        public long get(S source) throws Exception {
+        public long get(final S source) throws Throwable {
             switch (type) {
-                case TYPE_PRIMITIVE_LONG:
-                    return ((Number) method.invoke(source, EMPTY_ARGS)).longValue();
+                case TYPE_LONG_PRIMITIVE:
+                    return isMethodStatic ? (long) methodHandle.invokeExact() : (long) methodHandle.invokeExact(source);
                 case TYPE_LONG_NUMBER:
-                    Number longNumber = (Number) method.invoke(source, EMPTY_ARGS);
+                    final Number longNumber = invoke(source);
                     return longNumber == null ? 0 : longNumber.longValue();
                 case TYPE_MAP:
-                    Map<?, ?> map = (Map<?, ?>) method.invoke(source, EMPTY_ARGS);
+                    final Map<?, ?> map = invoke(source);
                     return map == null ? 0 : map.size();
                 case TYPE_COLLECTION:
-                    Collection<?> collection = (Collection<?>) method.invoke(source, EMPTY_ARGS);
+                    final Collection<?> collection = invoke(source);
                     return collection == null ? 0 : collection.size();
                 case TYPE_COUNTER:
-                    Counter counter = (Counter) method.invoke(source, EMPTY_ARGS);
+                    final Counter counter = invoke(source);
                     return counter == null ? 0 : counter.get();
                 case TYPE_SEMAPHORE:
-                    Semaphore semaphore = (Semaphore) method.invoke(source, EMPTY_ARGS);
+                    final Semaphore semaphore = invoke(source);
                     return semaphore == null ? 0 : semaphore.availablePermits();
                 default:
-                    throw new IllegalStateException("Unrecognized type:" + type);
+                    throw new IllegalStateException("Unrecognized type: " + type);
             }
         }
     }
 
     static class DoubleMethodProbe<S> extends MethodProbe implements DoubleProbeFunction<S> {
-
-        DoubleMethodProbe(Method method, Probe probe, int type, SourceMetadata sourceMetadata) {
+        DoubleMethodProbe(final Method method, final Probe probe, final ProbeType type, final SourceMetadata sourceMetadata) {
             super(method, probe, type, sourceMetadata);
         }
 
         @Override
-        public double get(S source) throws Exception {
+        public double get(final S source) throws Throwable {
             switch (type) {
                 case TYPE_DOUBLE_PRIMITIVE:
```

Comment:
I believe what is needed here is:

```

        @Override
        public double get(S source) throws Exception {
            switch (type) {
                case TYPE_PRIMITIVE_FLOAT:
                    return (float) METHOD_HANDLE.invokeExact(source);
                case TYPE_PRIMITIVE_DOUBLE:
                    return (double) METHOD_HANDLE.invokeExact(source);
                case TYPE_DOUBLE_NUMBER:
                    Number result = (Number) method.invoke(source, EMPTY_ARGS);
                    return result == null ? 0 : result.doubleValue();
                default:
                    throw new IllegalStateException("Unrecognized type:" + type);
            }
        }```

Reply:
See [previous comment](https://github.com/hazelcast/hazelcast/pull/25279#discussion_r1336864447).